### ISSUE TITLE
Google PubSub Subscription stops pulling messages from topic.

### ISSUE THREAD

**Issue by seanmt13** [GoogleCloudPlatform/google-cloud-python#5562]
In response to issue [5444 - Pubsub 0.35.3 stop consuming messages](https://github.com/GoogleCloudPlatform/google-cloud-python/issues/5444#issuecomment-400500096) and per theacodes request I am opening up a new issue.

I am currently running a container on kubernetes with 40 pods that are pulling from a subscription that is receiving ~50 messages a second. I am currently running 35.4

This is my code that i modified from the documentation that is pulling the messages.

```
def main():
    #Receives messages from a pull subscription.
    subscriber = pubsub_v1.SubscriberClient()
    subscription_path = subscriber.subscription_path(project, subscription_name)

    def callback(message):
        log.info('Received message: {}'.format(message))
        try:
            ret = bl.process_message(message)
            if ret.status_code == 200:
                log.info('200 File processed')
                statsd.increment('data-log-file-processed')
                message.ack()
            else:
                log.error('Could not process message for data log file publisher. {}' .format(ret.message))
                message.ack()
        except Exception as e:
            log.error('Could not process message: {} error occurred. {}' .format(ret.message, e))
    subscriber.subscribe(subscription_path, callback=callback)

    # The subscriber is non-blocking, so we must keep the main thread from
    # exiting to allow it to process messages in the background.
    log.info('Listening for messages on {}'.format(subscription_path))
    while True:
        time.sleep(60)
```

The SubsciberClient will pull messages for a time than just stop pulling them. After putting the logs to debug mode i am seeing the following:

[Log File](https://docs.google.com/spreadsheets/d/11DnGk_HchcCwOH_vKnd_UnZqs25qWAT4nDqoyDQjdqk/edit?usp=sharing)

Line 97 is the last message I received after that it seems to just go into an endless loop without ever pulling any messages. I know there are more messages to be pulled because my other pods are still pulling messages from the topic.

Please let me know if there is anything else needed to get this fixed or if it is something I am doing wrong. Thanks!

**Comment by seanmt13** [comment]
This acctually isn't the right log to post closing the issue until i can reproduce the issue.

**Comment by theacodes** [comment]
@seanmt13 actually if that's your code it does reveal the underlying issue: You need to wait on the subscription future to receive notifications when unrecoverable errors occur:

```python
def main():
    #Receives messages from a pull subscription.
    subscriber = pubsub_v1.SubscriberClient()
    subscription_path = subscriber.subscription_path(project, subscription_name)

    def callback(message):
        log.info('Received message: {}'.format(message))
        try:
            ret = bl.process_message(message)
            if ret.status_code == 200:
                log.info('200 File processed')
                statsd.increment('data-log-file-processed')
                message.ack()
            else:
                log.error('Could not process message for data log file publisher. {}' .format(ret.message))
                message.ack()
        except Exception as e:
            log.error('Could not process message: {} error occurred. {}' .format(ret.message, e))
    future = subscriber.subscribe(subscription_path, callback=callback)

    # The subscriber is non-blocking, so we must keep the main thread from
    # exiting to allow it to process messages in the background.
    log.info('Listening for messages on {}'.format(subscription_path))
    
    # Wait for the result on the future. If an unrecoverable error occurs, this will raise.
    future.result()
```

**Comment by seanmt13** [comment]
Thanks for pointing that out. I will fix that and see if any unrecoverable errors are raised.
